news-bundle/src/Resources/contao/modules/ModuleNews.php

```diff
@@ -184,7 +184,7 @@ protected function parseArticle($objArticle, $blnAddArchive=false, $strClass='',
 			if (null !== ($figure = $figureBuilder->buildIfResourceExists()))
 			{
 				// Rebuild with link to news article if none is set
-				if (!$figure->getLinkHref())
+				if (!$figure->getLinkHref() && $objTemplate->link != Environment::get('request'))
```

Comment:
```suggestion
				if (!$figure->getLinkHref() && $intCount !== 0)
```
`$intCount` is zero for the reader module and `1` or higher for the list module AFAIK.

Reply:
Not sure about this. At least with `link !== requestUri` you can kind of see what the code does ("do not link to the page I am already on") but `count !== 0` is not comprehensible without further knowledge.

And `count !== 0` will not work if the `event_full` template is used in a news list module.

Reply:
> but `count !== 0` is not comprehensible without further knowledge.

Agree, we could use a descriptive comment, or maybe another parameter for the method like `$blnForReaderPage`?

> And `count !== 0` will not work if the `event_full` template is used in a news list module.

I thought this is exactly what we want. No auto-link in the reader module, but still auto-link in the list module (even if the full template is used). This already works that way for events, see my other comment: https://github.com/contao/contao/pull/5851#discussion_r1133995625

To my understanding, only the news reader module behaves wrong currently, not the events reader.

Reply:
You are right.

Reply:
I still think `$objTemplate->link != Environment::get('request')` is the better choice, because it does what we actually want to do, which is not link to the page we are already on.

This way no one needs to know about `$intCount` and that is has to be set to link news images to the reader page, which is kind of a weird DX in my opinion. @contao/developers WDYT?